### ISSUE TITLE
Hipchat Import Error: request entity too large

### ISSUE THREAD

**Issue by tosinginchinese** [RocketChat/Rocket.Chat#13339]
Hello,
 
I was trying the Hipchat Importer in Rocket Chat 0.74.1 (Docker installation), I got an Hipchat export file and decrypted it with openssl aes-256-cbc -d -in hipchat-.....-2019-01-08_12-25-51.tar.gz.aes -out hipchat-.....-2019-01-08_12-25-51.tar.gz
It's a self-hosted instance and I'm not using Nginx as reverse proxy but HAProxy, same problem without a reverse proxy too.
I was able to import the file with 0.70.4 version, not with newer ones.
The file is 35,5MB
 
Importer’s log is the following:
 
rocketchat_1          | Error: request entity too large
rocketchat_1          |     at readStream (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/node_modules/raw-body/index.js:196:17)
rocketchat_1          |     at getRawBody (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/node_modules/raw-body/index.js:106:12)
rocketchat_1          |     at read (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/lib/read.js:68:3)
rocketchat_1          |     at jsonParser (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/lib/types/json.js:121:5)
rocketchat_1          |     at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40

Do you have a solution about it?

Thanks in advance

**Comment by Hudell** [comment]
This seems to be an issue with the file upload. If you can't get it to work, you can move the file to your rocket.chat server and use the file path to import instead of uploading it on the interface.

**Comment by mredbishop** [comment]
This may not be related but I have the same issue with a Slack import, my config is as follows:

![image](https://user-images.githubusercontent.com/1096293/53491434-52c63800-3a8e-11e9-9972-283d2052f713.png)

The error log is as follows:
```
root@chat:~# service snap.rocketchat-server.rocketchat-server status
● snap.rocketchat-server.rocketchat-server.service - Service for snap application rocketchat-server.rocketchat-server
   Loaded: loaded (/etc/systemd/system/snap.rocketchat-server.rocketchat-server.service; enabled; vendor preset: enabled)
   Active: active (running) since Wed 2019-02-27 12:20:10 UTC; 24min ago
 Main PID: 4753 (startRocketChat)
    Tasks: 11 (limit: 2362)
   CGroup: /system.slice/snap.rocketchat-server.rocketchat-server.service
           ├─4753 /bin/bash /snap/rocketchat-server/1364/bin/startRocketChat
           └─4806 node /snap/rocketchat-server/1364/main.js

Feb 27 12:27:44 chat rocketchat-server.rocketchat-server[4753]:     at read (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-
Feb 27 12:27:44 chat rocketchat-server.rocketchat-server[4753]:     at jsonParser (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules
Feb 27 12:27:44 chat rocketchat-server.rocketchat-server[4753]:     at /snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
Feb 27 12:35:20 chat rocketchat-server.rocketchat-server[4753]: Setting default file store to GridFS
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]: Error: request entity too large
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]:     at readStream (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]:     at getRawBody (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]:     at read (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]:     at jsonParser (/snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules
Feb 27 12:35:54 chat rocketchat-server.rocketchat-server[4753]:     at /snap/rocketchat-server/1364/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
```

**Comment by chirospasm** [comment]
> 
> 
> This seems to be an issue with the file upload. If you can't get it to work, you can move the file to your rocket.chat server and use the file path to import instead of uploading it on the interface.

Tested this on localhost server -- this unfortunately does not resolve the issue.

**Comment by Hudell** [comment]
@mredbishop since the code base for the importers are different, I'll ask you to open a separate issue for the slack importer so we can properly keep track of both problems.

@chirospasm can you enable debug level logs and try again, then share the logs here?

**Comment by chirospasm** [comment]
@Hudell , thanks -- I have opened this as a separate issue for Slack import today elsewhere, since I realized this related to Hipchat, and included logs there (#13628). Appreciate it!

**Comment by jby** [comment]
I have the same problem for HipChat, my file is 622MB zip (.tar.gz blatantly refuses to load, converting to zip produced an error at least (see below). I'm running in docker in MacOS 10.14. I have set my upload limit to '-1' and still get this. It did take a while though, I guess it worked on the file in 2-3 minutes before bailing out with this output in the logs from the docker container:

```
rocketchat_1  | Error: request entity too large
rocketchat_1  |     at readStream (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/node_modules/raw-body/index.js:196:17)
rocketchat_1  |     at getRawBody (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/node_modules/raw-body/index.js:106:12)
rocketchat_1  |     at read (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/lib/read.js:68:3)
rocketchat_1  |     at jsonParser (/app/bundle/programs/server/npm/node_modules/meteor/simple_json-routes/node_modules/connect/node_modules/body-parser/lib/types/json.js:121:5)
rocketchat_1  |     at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
```

**Comment by jby** [comment]
> This seems to be an issue with the file upload. If you can't get it to work, you can move the file to your rocket.chat server and use the file path to import instead of uploading it on the interface.

How would I do that then?
Just put it in the local filesystem and use that path in the file-upload dialog?

**Comment by Hudell** [comment]
@jby 
I'm not sure but I think the import process wouldn't know how to open a .zip file, it expects it to be a .tar.gz.

Instead you can extract the file on the local filesystem and use the path for the extracted folder on the "File URL" option of the import.

**Comment by jby** [comment]
From the adminpanel you can choose import -> HipChat and then either .tar.gz or .zip file.

I still don't understand what path I'm supposed to enter and where. I'm running in a docker container on MacOS, should it be a local path within the container and that local path in the webpage that asks for an URL to load from?

**Comment by jby** [comment]
Ok, yes - it **does** import if file is put in filesystem and the local path is used in the "File URL"!

**Comment by pierre-lehnen-rc** [comment]
Fixed in #16279
